### DISCUSSION TITLE
How to redirect www to non-www in nextjs 9.5 in next.config.js?

### DISCUSSION THREAD

**Discussion by isggwp** [vercel/next.js#17298] [Help] [answered]
its Possible redirect www to non-www url in Nextjs 9.5 version using next.config.js ?

**Comment by lfades** [comment, marked as answer]
This is currently out of the scope of Next.js, rewrites/headers/redirects only apply for relative paths, ignoring the domain.

**Reply by isggwp** [reply]
Ok, Thanks for the respons. can i use express for redirect like res.redirect on my custom server? it work on next 9.4 but getting error when i migrate to next 9.5

**Reply by lfades** [reply]
If you're using a custom server on top of Next.js you are free to use whatever tooling the custom server provides 👍

**Reply by isggwp** [reply]
thanks @lfades , now i rederect use nodejs http and url API at server.js. and it works

**Reply by Igor2122** [reply]
hey can u please share the code how did u handle the redirect ? 
thx

**Reply by MuhammadFaisalChughtai** [reply]
you can easily handle permanent redirection in nextjs using the below-mentioned code.

export const getServerSideProps = async (context) => {
  const { req, res } = context;
  if (req.headers.host.match(/^www/) !== null) {
    res.writeHead(301, {
      location: "https://" + req.headers.host.replace(/^www\./, "")  + req.url,
    });
    res.end();
  }
  return { props: {} };
};
